data_model/src/transaction.rs

```diff
@@ -405,12 +401,10 @@ mod candidate {
         }
 
         #[cfg(feature = "std")]
-        fn retain_verified_signatures(
-            &mut self,
-        ) -> Vec<&iroha_crypto::SignatureOf<TransactionPayload>> {
+        fn validate_signatures(&self) -> Result<(), &'static str> {
             self.signatures
-                .retain_verified_by_hash(self.payload.hash())
```

Comment:
The same pattern (removing invalid signatures) uses on the Sumeragi level. Shouldn't it be replaced there too?

Reply:
That's for blocks, not transactions. There it makes sense to tolerate invalid signatures